### PR TITLE
Added Azion DNS API

### PR DESCRIPTION
**Pull request by marcusgrando**
Add Azion DNS API.

Issue to reports bugs/improvements #3555

### CODE REVIEW

**Review by gabrielnes-azion — APPROVED** [review]

**Review by AndreSilveiraAzion — APPROVED** [review]

**Comment by Neilpang** [comment]
please check this guide.
https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by marcusgrando** [comment]
> please check this guide.
> https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide

@Neilpang I've created Issue and added on Wiki page. As I know, was only two steps I left behind. If no, can you be more specific?

File: dnsapi/dns_azion.sh [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+# Usage: add  _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+# Used to add txt record
+dns_azion_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  AZION_Username="${AZION_Username:-$(_readaccountconf_mutable AZION_Username)}"
+  AZION_Password="${AZION_Password:-$(_readaccountconf_mutable AZION_Password)}"
+  AZION_Token="${AZION_Token:-$(_readaccountconf_mutable AZION_Token)}"
+  AZION_ZoneID="${AZION_ZoneID:-$(_readaccountconf_mutable AZION_ZoneID)}"
```
Comment (Neilpang) [inline dnsapi/dns_azion.sh:23]: please remove the "AZION_ZoneID",   we should not need it.
And we can not save it in the account conf.

The user may issue certs for different zones everytime.

we should  get the zone id by REST api every time.

File: dnsapi/dns_azion.sh [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+# Usage: add  _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+# Used to add txt record
+dns_azion_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  AZION_Username="${AZION_Username:-$(_readaccountconf_mutable AZION_Username)}"
+  AZION_Password="${AZION_Password:-$(_readaccountconf_mutable AZION_Password)}"
+  AZION_Token="${AZION_Token:-$(_readaccountconf_mutable AZION_Token)}"
```
Comment (Neilpang) [inline dnsapi/dns_azion.sh:22]: how does the "AZION_Token" work?
Can we get it everytime or can we save it for future use?  How long will it be valid? Is it valid forever ?

File: dnsapi/dns_azion.sh (hunk 1 above)
Comment (marcusgrando) [reply dnsapi/dns_azion.sh]: Removed

File: dnsapi/dns_azion.sh (hunk 2 above)
Comment (marcusgrando) [reply dnsapi/dns_azion.sh]: Simplified code about Token. You are right, token expires and it can't be save.

File: dnsapi/dns_azion.sh [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      answers_list=$(echo "$_json_record" | _egrep_o "\"answers_list\":\[.*\]" | _head_n 1 | cut -d : -f 2 | tr -d \[\])
+      return 0
+    fi
+    return 1
+  fi
+  return 1
+}
+
+_get_token() {
+  AZION_Username="${AZION_Username:-$(_readaccountconf_mutable AZION_Username)}"
```
Comment (Neilpang) [inline dnsapi/dns_azion.sh:143]: do you think it's better to use a name like: `AZION_Email`  instead of `AZION_Username` ?

Comment (Neilpang) [inline dnsapi/dns_azion.sh:143]: do you think it's better to use a name like: `AZION_Email`  instead of `AZION_Username` ?

Comment (marcusgrando) [reply dnsapi/dns_azion.sh]: Yes, make sense. Updated and running tests again.

Comment (marcusgrando) [reply dnsapi/dns_azion.sh]: Yes, make sense. Updated and running tests again.

File: dnsapi/dns_azion.sh [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+_get_record() {
+  _domain_id=$1
+  _record=$2
+
+  if ! _azion_rest GET "intelligent_dns/$_domain_id/records"; then
+    return 1
+  fi
+
+  if _contains "$response" "\"entry\":\"$_record\""; then
+    _json_record=$(echo "$response" | tr '{}' "\n" | grep "\"entry\":\"$_record\"")
```
Comment (Neilpang) [inline dnsapi/dns_azion.sh:131]: please use double `\n` here:

```
tr '{}' "\n\n"
```

Comment (marcusgrando) [reply dnsapi/dns_azion.sh]: Hi @Neilpang, shellcheck doesn't like this:
`In dnsapi/dns_azion.sh line 131:
    _json_record=$(echo "$response" | tr '{}' "\n\n" | grep "\"entry\":\"$_record\"")
                                              ^----^ SC2020: tr replaces sets of chars, not words (mentioned due to duplicates).

For more information:
  https://www.shellcheck.net/wiki/SC2020`

Instead I changed to only one bracket. Running tests right now.

**Pull request merged**